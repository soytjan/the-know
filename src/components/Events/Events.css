@import url("https://fonts.googleapis.com/css?family=Open+Sans:300,400|Poiret+One");
.Events {
  margin: 0 auto;
  max-width: 1760px; }
  .Events button:hover {
    background-color: #BEECE9; }

.events-cont {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-row-gap: 20px;
  grid-column-gap: 15px;
  justify-items: center; }

.time-btn {
  border: none;
  font-size: 1.2em;
  font-weight: 400;
  background-color: #ECF0F1;
  width: 160px; }
  .time-btn:hover {
    cursor: pointer; }

.nav-time-container {
  width: 500px;
  display: flex;
  justify-content: space-between;
  margin: auto;
  padding: 0 0 25px; }
