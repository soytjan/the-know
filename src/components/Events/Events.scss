@import "../../variables.scss";

.Events {
  margin: 0 auto;
  max-width: 1760px;

  button {
    &:hover {
      background-color: $light-turq;
    }
  } 
}

.events-cont {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-row-gap: 20px;
  grid-column-gap: 15px;
  justify-items: center;
}

.time-btn {
  border: none;
  font-size: 1.2em;
  font-weight: 400;
  background-color: $main-bg;
  width: 160px;
  color: $turq;

  &:hover {
    cursor: pointer;
  }
}

.nav-time-container {
  width: 500px;
  display: flex;
  justify-content: space-between;
  margin: auto;
  padding: 0 0 25px;
}

@media screen and (max-width: 1530px) {
  .Events {
    max-width: 1190px;
  }

  .events-cont {
    grid-template-columns: repeat(4, 1fr);
  }
}

@media screen and (max-width: 1225px) {
  .Events {
    max-width: 900px;
  }

  .events-cont {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (max-width: 915px) {
  .Events {
    max-width: 670px;
  }

  .events-cont {
    grid-template-columns: repeat(2, 1fr);
  }
}


