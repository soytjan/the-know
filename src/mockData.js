export const mockEventData = {
  "last_item": null,
  "total_items": "9621",
  "first_item": null,
  "page_number": "1",
  "page_size": "10",
  "page_items": null,
  "search_time": "0.049",
  "page_count": "963",
  "events": {
    "event": [
      {
        "watching_count": null,
        "olson_path": "America/Denver",
        "calendar_count": null,
        "comment_count": null,
        "region_abbr": "CO",
        "postal_code": null,
        "going_count": null,
        "all_day": "0",
        "latitude": "39.7392358",
        "groups": null,
        "url": "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
        "id": "E0-001-111209195-8",
        "privacy": "1",
        "city_name": "Denver",
        "link_count": null,
        "longitude": "-104.990251",
        "country_name": "United States",
        "country_abbr": "USA",
        "region_name": "Colorado",
        "start_time": "2018-07-05 09:00:00",
        "tz_id": null,
        "description": " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
        "modified": "2018-02-19 06:12:10",
        "venue_display": "0",
        "tz_country": null,
        "performers": null,
        "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
        "venue_address": "Denver, Colorado, United States",
        "geocode_type": "EVDB Geocoder",
        "tz_olson_path": null,
        "recur_string": null,
        "calendars": null,
        "owner": "evdb",
        "going": null,
        "country_abbr2": "US",
        "image": null,
        "created": "2018-02-01 04:56:32",
        "venue_id": "V0-001-000179395-7",
        "tz_city": null,
        "stop_time": null,
        "venue_name": "Denver, Colorado, United States",
        "venue_url": "http://denver.eventful.com/venues/denver-colorado-united-states-/V0-001-000179395-7?utm_source=apis&utm_medium=apim&utm_campaign=apic"
      }
    ]
  }
}

export const mockCleanEventData = [
  {
    title: "Thursday Work Day, July 5 - Multiple Neighborhoods",
    description: " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
    category: "event",
    venueName: "Denver, Colorado, United States", 
    region: "CO",
    postalCode: null,
    city: "Denver",
    startTime: "2018-07-05 09:00:00",
    venueAddress: "Denver, Colorado, United States",
    image: null,
    url: "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
    id: "E0-001-111209195-8"
  }
]