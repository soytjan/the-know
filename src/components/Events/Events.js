import React, { Component } from 'react';
// import { connect } from 'react-redux';
// import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import EventCard from '../EventCard/EventCard';
import './Events.css';

class Events extends Component {
  genEventsArray = () => {
    const { info, type, onFavorite } = this.props;
    let events;
    if (type === 'event') {
      const categories = Object.keys(info);
      //think about a refactor here
      // const masterEvents = categories.reduce((obj, key) => {
      //   console.log('category objects', info[key])
      // }, {});
      // console.log(masterEvents, 'masterEvents')
      return categories.reduce((eventsArr, category) => {
        const categoryIds = Object.keys(info[category]);
        const categoryEvents = categoryIds.map(eventId => info[category][eventId]);

        return [...eventsArr, ...categoryEvents]
      }, [])
    } 
      events = info;
      const categoryIds = Object.keys(events);
      return categoryIds.map(eventId => events[eventId])
  }

  render() {
    const { info, type, onFavorite } = this.props;
    const eventArray = this.genEventsArray()
    const renderedEvents = eventArray.map(event => {
      return <EventCard event={event} type={type} onFavorite={onFavorite} /> 
    })

    return (
      <section className='Events'>
        I'm the Events!
        { renderedEvents }
      </section>
    )
  }
}

Events.propTypes = {
  info: PropTypes.array,
  type: PropTypes.string,
  onFavorite: PropTypes.func
};

export default Events;