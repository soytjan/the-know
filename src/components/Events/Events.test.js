import React from 'react';
import { shallow } from 'enzyme';
import { mockCleanEventData, mockEventDataArray } from '../../mockData';
import Events from './Events';

describe('Events', () => {
  let renderedComponent;
  let mockInfo;
  let mockOnFavorite;
   
  beforeAll(() => {
    mockInfo = {
      '1': {title: 'event1', category: 'music', startTime: "2018-07-05 09:00:00", id: '1'}, 
      '2': {title: 'event2', category: 'food', startTime: "2018-02-28 09:00:00", id: '2'}, 
      '3': {title: 'event3', category: 'culture', startTime: "2018-03-20 09:00:00", id: '3'}
    };
  });

  beforeEach(() => {
    mockOnFavorite = jest.fn();
    renderedComponent = shallow(
      <Events 
        info={mockInfo}
        type='music'
        onFavorite={mockOnFavorite}
      />
    );
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
  
  describe('convertEvents', () => {
    it('should return an array of objects with all the event objects', () => {
      const result = renderedComponent.instance().convertEvents();
      const expected = [{title: 'event1', category: 'music', startTime: "2018-07-05 09:00:00", id: '1'}, {title: 'event2', category: 'food', startTime: "2018-02-28 09:00:00", id: '2'}, {title: 'event3', category: 'culture', startTime: "2018-03-20 09:00:00", id: '3'}];
      
      expect(result).toEqual(expected);
    });

    it('should return an array of objects with all the events when a type of event are passed in', () => {
      const mockInfo = {music: mockCleanEventData};
      const renderedComponent = shallow(
        <Events 
          info={mockInfo}
          type='event'
          onFavorite={mockOnFavorite}
        />
      );
      const result = renderedComponent.instance().convertEvents();
      const expected = mockEventDataArray;

      expect(result).toEqual(expected);
    });
  });

  describe('filterEvents', () => {
    it('should return only events that are this week when time state is week', () => {
      const expected = [{title: 'event2', category: 'food', startTime: "2018-02-28 09:00:00", id: '2'}];

      renderedComponent.setState({ time: 'week' });
      expect(renderedComponent.instance().filterEvents()).toEqual(expected);
    });

    it('should return only events that are this month when time state is month', () => {
      const expected = [
        {title: 'event2', category: 'food', startTime: "2018-02-28 09:00:00", id: '2'}, 
        {title: 'event3', category: 'culture', startTime: "2018-03-20 09:00:00", id: '3'}
      ];

      renderedComponent.setState({ time: 'month' });

      expect(renderedComponent.instance().filterEvents()).toEqual(expected);
    });

    it('should return all events when time state is all', () => {
      const expected = [
        {title: 'event1', category: 'music', startTime: "2018-07-05 09:00:00", id: '1'}, 
        {title: 'event2', category: 'food', startTime: "2018-02-28 09:00:00", id: '2'}, 
        {title: 'event3', category: 'culture', startTime: "2018-03-20 09:00:00", id: '3'}
      ];
      
      expect(renderedComponent.instance().filterEvents()).toEqual(expected);
    });
  });

  describe('handleClick', () => {
    it('should set time state to week when THIS WEEK button is clicked', () => {
      expect(renderedComponent.state('time')).toEqual('all');

      renderedComponent.instance().handleClick('week');

      expect(renderedComponent.state('time')).toEqual('week');
    });

    it('should set time state to week when THIS MONTH button is clicked', () => {
      expect(renderedComponent.state('time')).toEqual('all');

      renderedComponent.instance().handleClick('month');

      expect(renderedComponent.state('time')).toEqual('month');
    });

    it('should set time state to week when THIS MONTH button is clicked', () => {
      expect(renderedComponent.state('time')).toEqual('all');

      renderedComponent.instance().handleClick('all');

      expect(renderedComponent.state('time')).toEqual('all');
    });
  });
});