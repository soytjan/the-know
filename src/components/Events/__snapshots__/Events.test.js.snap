// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Events
    info={
        Object {
            "1": Object {
              "category": "music",
              "id": "1",
              "startTime": "2018-07-05 09:00:00",
              "title": "event1",
            },
            "2": Object {
              "category": "food",
              "id": "2",
              "startTime": "2018-02-28 09:00:00",
              "title": "event2",
            },
            "3": Object {
              "category": "culture",
              "id": "3",
              "startTime": "2018-03-20 09:00:00",
              "title": "event3",
            },
          }
    }
    onFavorite={[Function]}
    type="music"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="nav-time-container"
>
          <button
                    className="time-btn"
                    onClick={[Function]}
          >
                    THIS WEEK
          </button>
          <button
                    className="time-btn"
                    onClick={[Function]}
          >
                    THIS MONTH
          </button>
          <button
                    className="time-btn"
                    onClick={[Function]}
          >
                    ALL UPCOMING
          </button>
</nav>,
        <section
          className="events-cont"
>
          <EventCard
                    event={
                              Object {
                                        "category": "music",
                                        "id": "1",
                                        "startTime": "2018-07-05 09:00:00",
                                        "title": "event1",
                                      }
                    }
                    onFavorite={[Function]}
          />
          <EventCard
                    event={
                              Object {
                                        "category": "food",
                                        "id": "2",
                                        "startTime": "2018-02-28 09:00:00",
                                        "title": "event2",
                                      }
                    }
                    onFavorite={[Function]}
          />
          <EventCard
                    event={
                              Object {
                                        "category": "culture",
                                        "id": "3",
                                        "startTime": "2018-03-20 09:00:00",
                                        "title": "event3",
                                      }
                    }
                    onFavorite={[Function]}
          />
</section>,
      ],
      "className": "Events",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="time-btn"
              onClick={[Function]}
>
              THIS WEEK
</button>,
            <button
              className="time-btn"
              onClick={[Function]}
>
              THIS MONTH
</button>,
            <button
              className="time-btn"
              onClick={[Function]}
>
              ALL UPCOMING
</button>,
          ],
          "className": "nav-time-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "THIS WEEK",
              "className": "time-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "THIS WEEK",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "THIS MONTH",
              "className": "time-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "THIS MONTH",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ALL UPCOMING",
              "className": "time-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "ALL UPCOMING",
            "type": "button",
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <EventCard
              event={
                            Object {
                                          "category": "music",
                                          "id": "1",
                                          "startTime": "2018-07-05 09:00:00",
                                          "title": "event1",
                                        }
              }
              onFavorite={[Function]}
/>,
            <EventCard
              event={
                            Object {
                                          "category": "food",
                                          "id": "2",
                                          "startTime": "2018-02-28 09:00:00",
                                          "title": "event2",
                                        }
              }
              onFavorite={[Function]}
/>,
            <EventCard
              event={
                            Object {
                                          "category": "culture",
                                          "id": "3",
                                          "startTime": "2018-03-20 09:00:00",
                                          "title": "event3",
                                        }
              }
              onFavorite={[Function]}
/>,
          ],
          "className": "events-cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "event": Object {
                "category": "music",
                "id": "1",
                "startTime": "2018-07-05 09:00:00",
                "title": "event1",
              },
              "onFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "event": Object {
                "category": "food",
                "id": "2",
                "startTime": "2018-02-28 09:00:00",
                "title": "event2",
              },
              "onFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "event": Object {
                "category": "culture",
                "id": "3",
                "startTime": "2018-03-20 09:00:00",
                "title": "event3",
              },
              "onFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="nav-time-container"
>
            <button
                        className="time-btn"
                        onClick={[Function]}
            >
                        THIS WEEK
            </button>
            <button
                        className="time-btn"
                        onClick={[Function]}
            >
                        THIS MONTH
            </button>
            <button
                        className="time-btn"
                        onClick={[Function]}
            >
                        ALL UPCOMING
            </button>
</nav>,
          <section
            className="events-cont"
>
            <EventCard
                        event={
                                    Object {
                                                "category": "music",
                                                "id": "1",
                                                "startTime": "2018-07-05 09:00:00",
                                                "title": "event1",
                                              }
                        }
                        onFavorite={[Function]}
            />
            <EventCard
                        event={
                                    Object {
                                                "category": "food",
                                                "id": "2",
                                                "startTime": "2018-02-28 09:00:00",
                                                "title": "event2",
                                              }
                        }
                        onFavorite={[Function]}
            />
            <EventCard
                        event={
                                    Object {
                                                "category": "culture",
                                                "id": "3",
                                                "startTime": "2018-03-20 09:00:00",
                                                "title": "event3",
                                              }
                        }
                        onFavorite={[Function]}
            />
</section>,
        ],
        "className": "Events",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="time-btn"
                onClick={[Function]}
>
                THIS WEEK
</button>,
              <button
                className="time-btn"
                onClick={[Function]}
>
                THIS MONTH
</button>,
              <button
                className="time-btn"
                onClick={[Function]}
>
                ALL UPCOMING
</button>,
            ],
            "className": "nav-time-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "THIS WEEK",
                "className": "time-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "THIS WEEK",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "THIS MONTH",
                "className": "time-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "THIS MONTH",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ALL UPCOMING",
                "className": "time-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "ALL UPCOMING",
              "type": "button",
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <EventCard
                event={
                                Object {
                                                "category": "music",
                                                "id": "1",
                                                "startTime": "2018-07-05 09:00:00",
                                                "title": "event1",
                                              }
                }
                onFavorite={[Function]}
/>,
              <EventCard
                event={
                                Object {
                                                "category": "food",
                                                "id": "2",
                                                "startTime": "2018-02-28 09:00:00",
                                                "title": "event2",
                                              }
                }
                onFavorite={[Function]}
/>,
              <EventCard
                event={
                                Object {
                                                "category": "culture",
                                                "id": "3",
                                                "startTime": "2018-03-20 09:00:00",
                                                "title": "event3",
                                              }
                }
                onFavorite={[Function]}
/>,
            ],
            "className": "events-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "event": Object {
                  "category": "music",
                  "id": "1",
                  "startTime": "2018-07-05 09:00:00",
                  "title": "event1",
                },
                "onFavorite": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "event": Object {
                  "category": "food",
                  "id": "2",
                  "startTime": "2018-02-28 09:00:00",
                  "title": "event2",
                },
                "onFavorite": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "function",
              "props": Object {
                "event": Object {
                  "category": "culture",
                  "id": "3",
                  "startTime": "2018-03-20 09:00:00",
                  "title": "event3",
                },
                "onFavorite": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
